rs/state_manager/src/lib.rs

```diff
@@ -3248,6 +3261,9 @@ impl StateManager for StateManagerImpl {
             self.metrics
                 .latest_certified_height
                 .set(latest_certified as i64);
+            self.metrics
```

Comment:
I'm not quite sure how to interpret this metric in the end. The first replica finishing the DSM will report the largest value. But 1/3 of replicas would essentially report 0, as the certification would already be in the consensus pool.

Reply:
My thinking with all these metrics is that the median across the subnet should be meaningful (likely also the mean, come to think of it).

Do you think that is not the case here?

(I agree that individual replicas' numbers may be skewed, with fast replicas showing higher latency than slow replicas. It may even make sense to take a higher percentile than the median, e.g. 67th, which would fall in the middle of the first `2f+1` replicas.)

Reply:
As seen today while investigating the high load on `lhg73`, because the subnet has about half Gen1 and half Gen2 replicas; and because the Gen1 nodes appear to be significantly slower under load than Gen2 nodes; when some replicas' DSMs are significantly ahead of other replicas' DSMs, you will end up with the former showing very large certification latencies (multiple seconds) while for the latter certification is almost instant.

In such a case, both the median replica's and the 67th percentile replica's latency Would be meaningful. The 67th percentile (as long as it's used consistently across all stages that a block / message goes through) would more or less illustrate the behavior of the marginal (`2f+1` slowest) replica. While the median (whether in the average case; or in an extreme situation where half the replicas are fast, half are slow) will illustrate the behavior of some arbitrary replica that's part of the quorum. Whether that's a "fast" replica, with most of its latency coming from MR waiting for the next block and certification; or a "slow" replica, with most of its latency observed in induction and processing. Regardless, the E2E latency should be the same.